[BUG]: Typescript error on paginate string input

### What happened?

When using `octokit.paginate`, Typescript errors saying (I think?) that it does not accept a string as the first input. My understanding based on the docs and @octokit/types code is that this input can be a string, and I have confirmed that the code builds and runs if I ignore this error with `// @ts-ignore`. I initially had this same code but using `octokit.request` instead and did not have the error. 


```typescript
const alerts: Array<DependabotAlert> = await octokit.paginate(
  "GET /orgs/{org}/dependabot/alerts",
  {
    org: GITHUB_ORG,
    per_page: 100,
    headers: {
      "X-GitHub-Api-Version": "2022-11-28",
    },
  },
  (response) =>
    response.data.map(
      (item) =>
        <DependabotAlert>{
          createdAt: item.created_at,
          cve: item.security_advisory.cve_id,
          ecosystem: item.security_vulnerability.package.ecosystem,
          number: item.number,
          package: item.security_vulnerability.package.name,
          repository: item.repository.name,
          severity: item.security_vulnerability.severity,
          state: item.state,
          url: item.html_url,
        }
    )
);
```
_Note: `DependabotAlert` is my own custom type that defines the resulting output._


**What happened:** Typescript error when building

```
No overload matches this call.
...
Argument of type 'string' is not assignable to parameter of type 'RequestInterface<object>'. ts(2769)
```

**What I expect to happen:** No Typescript error. Should build and run with `"GET /orgs/{org}/dependabot/alerts"` (or any string) as input.

**Workaround:** Add `// @ts-ignore` above this function call so that Typescript will ignore it and build anyways. 

### Versions

nodejs 16.20.0
octokit 2.0.14
@octokit/types 9.0.0 (as a sub-dependency of octokit)
typescript 5.0.4

### Relevant log output

```shell
error TS2769: No overload matches this call.
  Overload 1 of 9, '(route: "GET /orgs/{org}/dependabot/alerts", parameters: { org: string; } & { state?: string | undefined; severity?: string | undefined; ecosystem?: string | undefined; package?: string | undefined; ... 7 more ...; per_page?: number | undefined; }, mapFn: MapFunction<...>): Promise<...>', gave the following error.
    Argument of type '{ org: string; per_page: number; headers: { "X-GitHub-Api-Version": string; }; }' is not assignable to parameter of type '{ org: string; } & { state?: string | undefined; severity?: string | undefined; ecosystem?: string | undefined; package?: string | undefined; scope?: "development" | "runtime" | undefined; ... 6 more ...; per_page?: number | undefined; }'.
      Object literal may only specify known properties, and 'headers' does not exist in type '{ org: string; } & { state?: string | undefined; severity?: string | undefined; ecosystem?: string | undefined; package?: string | undefined; scope?: "development" | "runtime" | undefined; ... 6 more ...; per_page?: number | undefined; }'.
  Overload 2 of 9, '(request: RequestInterface<object>, parameters: string, mapFn: MapFunction<NormalizeResponse<OctokitResponse<any>>, DependabotAlert[]>): Promise<...>', gave the following error.
    Argument of type 'string' is not assignable to parameter of type 'RequestInterface<object>'.

  9   const alerts: Array<DependabotAlert> = await octokit.paginate(
                                                   ~~~~~~~~~~~~~~~~~
 10     "GET /orgs/{org}/dependabot/alerts",
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
... 
 32       )
    ~~~~~~~
 33   );
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

@wolfy1339 still having an issue. Building with `tsc` results in similar but not exactly the same error as before:

```
error TS2769: No overload matches this call.
Overload 1 of 9, '(route: "GET /orgs/{org}/dependabot/alerts", parameters: { state?: string | undefined; severity?: string | undefined; ecosystem?: string | undefined; package?: string | undefined; scope?: "development" | ... 1 more ... | undefined; ... 6 more ...; per_page?: number | undefined; } & { ...; }, mapFn: MapFunction<...>): Promise<...>', gave the following error.
    Argument of type '{ org: string; per_page: number; headers: { "X-GitHub-Api-Version": string; }; state: string; }' is not assignable to parameter of type '{ state?: string | undefined; severity?: string | undefined; ecosystem?: string | undefined; package?: string | undefined; scope?: "development" | "runtime" | undefined; sort?: "created" | "updated" | undefined; ... 5 more ...; per_page?: number | undefined; } & { ...; }'.
      Object literal may only specify known properties, and 'headers' does not exist in type '{ state?: string | undefined; severity?: string | undefined; ecosystem?: string | undefined; package?: string | undefined; scope?: "development" | "runtime" | undefined; sort?: "created" | "updated" | undefined; ... 5 more ...; per_page?: number | undefined; } & { ...; }'.
  Overload 2 of 9, '(request: RequestInterface<object>, parameters: string, mapFn: MapFunction<NormalizeResponse<OctokitResponse<any>>, DependabotAlert[]>): Promise<...>', gave the following error.
    Argument of type 'string' is not assignable to parameter of type 'RequestInterface<object>'.

  9   const alerts: Array<DependabotAlert> = await octokit.paginate(
                                                   ~~~~~~~~~~~~~~~~~
 10     "GET /orgs/{org}/dependabot/alerts",
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
... 
 34       )
    ~~~~~~~
 35   );
```

Code is the same as before, except for the addition of the `state` property:

```typescript
const alerts: Array<DependabotAlert> = await octokit.paginate(
  "GET /orgs/{org}/dependabot/alerts",
  {
    org: GITHUB_ORG,
    per_page: 100,
    headers: {
      "X-GitHub-Api-Version": "2022-11-28",
    },
    state: "open"
  },
  (response) =>
    response.data.map(
      (item) =>
        <DependabotAlert>{
          ...
        }
    )
);
```


Versions based on updated `package-lock.json`:
- node_modules/typescript: `5.0.4` (same as previously)
- node_modules/octokit: `2.0.14` (same as previously)
- node_modules/@octokit/types: `9.2.2` (was 9.0.0 previously)
- node_modules/@octokit/openapi-types: `17.1.2` (was 16.0.0 previously)

This is not a huge deal for me btw. I've just ignored this line with `// @ts-ignore`. But happy to continue to support as much as you like.

--- a/package.json
+++ b/package.json
@@ -20,14 +20,14 @@
     "test:typescript": "npx tsc --noEmit --declaration --noUnusedLocals --esModuleInterop --strict test/typescript-validate.ts"
   },
   "dependencies": {
-    "@octokit/app": "^13.1.1",
-    "@octokit/core": "^4.0.4",
-    "@octokit/oauth-app": "^4.0.6",
-    "@octokit/plugin-paginate-rest": "^6.0.0",
-    "@octokit/plugin-rest-endpoint-methods": "^7.0.0",
-    "@octokit/plugin-retry": "^4.0.3",
-    "@octokit/plugin-throttling": "^5.0.0",
-    "@octokit/types": "^9.0.0"
+    "@octokit/app": "^13.1.3",
+    "@octokit/core": "^4.2.1",
+    "@octokit/oauth-app": "^4.2.1",
+    "@octokit/plugin-paginate-rest": "^6.1.0",
+    "@octokit/plugin-rest-endpoint-methods": "^7.1.1",
+    "@octokit/plugin-retry": "^4.1.3",
+    "@octokit/plugin-throttling": "^5.2.2",
+    "@octokit/types": "^9.2.2"
   },
   "devDependencies": {
     "@octokit/tsconfig": "^1.0.2",

